### PR TITLE
Calendar Export

### PR DESCRIPTION
**Pull request by danakim21**
### Summary 

Changed side-view so that the circle button is below time and calendar export icon is added next to the name of the office hour. When a user clicks on the calendar export icon, a popup is opened. This shows two options -- Export to Google Calendar and Export to Apple Calendar. When Export to Google Calendar is clicked, a new event is added to the user's primary google calendar. When Export to Apple Calendar is clicked, an .ics file is downloaded so that the user can import it to their Apple Calendar. 

Screen Shot 2021-11-23 at 9 12 31 PM

Screen Shot 2021-11-23 at 9 15 13 PM

### Test Plan 

- [ ] Test by exporting both Google and Apple Calendar for: 
  - [ ] TA view 
  - [ ] Professor view 
  - [ ] Student view 
- [ ] Check that location is correct for: 
  - [ ] Virtual 
  - [ ] Hybrid 
  - [ ] In-person 
- [ ] Check that other information is correct (Name, Time, Date)

### Breaking Changes 

None

- [ ] I have updated the documentation accordingly.
- [ ] My PR adds a @ts-ignore

### CODE REVIEW

**Review by triple-jay — CHANGES_REQUESTED** [review]
Thank you so so much for working on this Dana!! This is definitely a **very challenging** task that I assigned you, and I appreciate you taking the initiative to learn a lot of this stuff on your own! I did test that the Apple Calendar export works as expected by clicking on the button to download the .ics file and dragging it into my calendar on my computer. However, the Google Calendar button does not work yet as far as I could tell. I know that you said you weren't finished with the PR when you messaged me, so it's totally fine if you didn't implement it yet :). Just letting you in the case that you did implement it so I could let you know it doesn't work yet on my end!

Regardless, thanks again for working on this, and **especially** for taking on the extra challenge to implement this with the Google API! It really makes me happy that you chose to go for the extra challenge because you thought it would be better for our users.

**Review by pooja654 — COMMENTED** [review]
This is really cool Dana! Your implementation looks really clean. I tested that the Apple Calendar export works (I successfully downloaded and added it to my calendar), and as Jeremy said, I noticed the Google Calendar one did not seem implemented yet.

One suggestion I have is to make the popup a little smaller and reduce the whitespace - I feel like it would look better.

**Review by nada-attia — COMMENTED** [review]
dana, good job with this pr! one suggestion: can you make the modal responsive on smaller screens? if u do cmd+option+j (assuming ur using chrome) and you click on the little phone icon you can see what the modal looks like on an iphone screen rn (its really big and you cant click the x button). 

you can refer to other existing modals when fixing this issue such as the popup that appears when you click leave queue. 

other than that great work dana :)

File: src/components/includes/CalendarExportModal.tsx [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    //         '11/15/2021',
+    //         '10:00 AM',
+    //         '05/30/2020',
+    //         '1:00 PM',
+    //         '50 multiple choice',
+    //         '106 Valentine Place',
+    //     ];
+    //     csvRows.push(values.join(','));
+    //     return csvRows.join('\n');
+    // };
```
Comment (nada-attia) [inline src/components/includes/CalendarExportModal.tsx:49]: also make sure u delete comments before merging

**Review by nada-attia — COMMENTED** [review]
dana, good job with this pr! one suggestion: can you make the model responsive on smaller screens? i u do cmd+option+j and you click on the little phone icon you can see how the modal looks like on an iphone screen rn (its really big and you cant click the x button). 

you can refer to other modals when fixing this issue such as the popup that appears when you click leave queue. 

other than that great work dana :)

**Pull request closed without merging**